### DISCUSSION TITLE
How to create a rigid body?

### DISCUSSION THREAD

**Discussion by Wang-Yihu** [idaholab/moose#22750] [Q&A Modules: Solid mechanics] [answered]
How to create a rigid body in MOOSE tensor mechanics? I try to make a material of 1000e9 Pa youngs_modulus or 0.4999 poissons_ratio, but it cannot get converge.
How to create a rigid body in MOOSE Multiphysics?

**Comment by GiudGiud** [comment]
@bwspenc @recuero

**Comment by bwspenc** [comment, marked as answer]
We don't currently have a rigid body capability. This would be a fairly major undertaking, and we don't really have a need for this for currently funded projects, so I don't anticipate that this is something we would develop in the near future.

In the mean time, I think your best bet is to set the Young's modulus to a high value as you suggested. Setting the Poisson's ratio to something close to 0.5 won't make the block rigid -- it will just make it nearly incompressible and make the problem ill conditioned. I guess I don't know what the stiffnesses of your other materials is, so it's hard to say whether the high Young's modulus you suggested would be an issue. You'll generally have better luck converging problems with poor conditioning using LU rather than iterative preconditioners. Using automatic scaling can help too. These settings in your Executioner block might be helpful, although I can't say for sure since I haven't seen the rest of the input:
```
  automatic_scaling = true
  compute_scaling_once = true
  petsc_options_iname = '-pc_type -pc_factor_mat_solver_package'
  petsc_options_value = ' lu       superlu_dist
